Sources/MemberwiseInitializerMacros/MemberwiseInitializerMacro.swift

```diff
@@ -20,12 +20,33 @@ public struct MemberwiseInitializerMacro: MemberMacro {
       .flatMap(\.bindings)
       .filter { $0.accessorBlock == nil }
 
-    let parameters = bindings.map { "\($0.pattern): \($0.typeAnnotation!.type)" }.joined(separator: ", ")
-    let header = SyntaxNodeString(stringLiteral: "public init(\(parameters))")
+    var parameters = [String]()
+    var initializations = [String]()
+    
+      bindings.forEach {
+      guard let typeAnnotation = $0.typeAnnotation else { return }
+      var typeAnnotationText = "\(typeAnnotation.type)"
+
+      // Detect if the property is a closure
+      if typeAnnotationText.contains("() ->") && !typeAnnotationText.contains("@escaping") {
+        typeAnnotationText = "@escaping " + typeAnnotationText
+      }
+            
+      // Prepare parameter string, auto-nil for optional values
+      let defaultValue = (typeAnnotationText.contains("?") || typeAnnotationText.contains("= nil")) ? " = nil" : ""
```

Comment:
@Lutzifer  I woudl also like to have your opionion on that. 

I understand that this solves a problem for one of our projects. With the change regarding closures i am pretty fine, but regarding the "auto-nil"-defaultvalue i am not sure if that always makes sense or creates some kind of trap for us as an uninitialized value is not complained anymore as soon as it is optional

Reply:
I second that.

@im4xpro 

- restore the original test, we should not have to change this to ensure backwards-compatibilty
- add a new test that documents what changes we need
- implement the macro so that both cases are successfull